x-pack/filebeat/tests/integration/input_metrics_test.go

```diff
@@ -0,0 +1,292 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"strings"
+	"testing"
+	"text/template"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInputMetricsFromPipeline_Filestream(t *testing.T) {
+	var tmplCfg = `
+http:
+  enabled: true
+filebeat.inputs:
+  - type: filestream
+    id: {{.filestream_id}}
```

Comment:
Is there an already existing test for the case where there is no ID? I see in the metrics implementation it doesn't create input metrics if there is no ID, but it would be good to have something to confirm that the Beat doesn't panic in this case.

The only time we are guaranteed to have a non-empty ID is when the input comes from a Fleet managed agent. Standalone Beats inputs do not require IDs at all, Beat inputs run by agent can have empty IDs as long as there is only a single instance of an input.

We don't necessarily need to duplicate this integration test, I am just doubling checking there is a test ensuring that these cases (which will absolutely happen in real life) behave reasonably.

Reply:
done here https://github.com/elastic/beats/pull/42618/commits/3cbde19a9d582406f41cdc9d74c8e759b58aa839